### Model Validation Technique
There are 3 Technique

1.Train & Test Split

2.K-fold Cross Validation

3.Leave One Out Cross Validation

#### Using a Logistic Model 

#### 1.Evaluate using a train and a test set

In [1]:
# Evaluate using a train and a test set

#Importing the Required Libraries
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression

import warnings
warnings.filterwarnings("ignore")  #--to ignore warnings

#Load the data set
claimants = pd.read_csv("C:/Users/Akaash/Downloads/claimants.csv")

#Dropping the case number columns as it is not required
claimants.drop(["CASENUM"],inplace=True,axis = 1)

# Removing NA values in data set
claimants = claimants.dropna()

# Dividing our data into input and output variables 
X = claimants.iloc[:,1:]
Y = claimants.iloc[:,0]

#Splitting the Dataset into Train & Test 
test_size = 0.33
seed = 7
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=test_size,random_state=seed)

#Training the Model
model = LogisticRegression()
model.fit(X_train, Y_train)

#Predicting & Getting the Model Score
result = model.score(X_test, Y_test)

#Calculating the Accuracy of the Model
result*100.0

71.27071823204419

Inference: With Train & Test Split The Accuarcy has come 71.27%

#### 2.Evaluate using K-Fold Cross Validation

In [2]:
# Evaluate using Cross Validation

#Importing the Required Libraries
import pandas as pd
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression

#Load the data set
claimants = pd.read_csv("C:/Users/Akaash/Downloads/claimants.csv")

#Dropping the case number columns as it is not required
claimants.drop(["CASENUM"],inplace=True,axis = 1)

# Removing NA values in data set
claimants = claimants.dropna()

# Dividing our data into input and output variables 
X = claimants.iloc[:,1:]
Y = claimants.iloc[:,0]

# Creating K-Fold Instance
num_folds = 10
seed = 7
kfold = KFold(n_splits=num_folds, random_state=seed,shuffle=True)

#Training the Model 
model = LogisticRegression(max_iter=400)

#Predicting & Getting the Model Score with kfold
results = cross_val_score(model, X, Y, cv=kfold)


In [3]:
# Accuracy for Each K-Fold
results

array([0.73636364, 0.70909091, 0.62727273, 0.7       , 0.70909091,
       0.69090909, 0.66972477, 0.70642202, 0.72477064, 0.73394495])

In [4]:
#Average of All k-fold Accuracy to get Final model Accuary
results.mean()*100.0

70.07589658048373

Inference: The K-fold gave the final Accuracy of 70.07%

#### 3.Evaluate using Leave One Out Cross Validation

In [5]:
# Evaluate using Leave One Out Cross Validation

#Importing the Required Libraries
import pandas as pd
from sklearn.model_selection import LeaveOneOut
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression

#Load the data set
claimants = pd.read_csv("C:/Users/Akaash/Downloads/claimants.csv")

#Dropping the case number columns as it is not required
claimants.drop(["CASENUM"],inplace=True,axis = 1)

# Removing NA values in data set
claimants = claimants.dropna()

# Dividing our data into input and output variables 
X = claimants.iloc[:,1:]
Y = claimants.iloc[:,0]

# Creating K-Fold Instance
loocv = LeaveOneOut()

#Training the Model
model = LogisticRegression(max_iter=300)

#Predicting & Getting the Model Score with Leave One Out Cross Validation
results = cross_val_score(model, X, Y, cv=loocv)

In [6]:
#Average of All LooCV Accuracy to get Final model Accuary
results.mean()*100.0

70.25547445255475

Inference: With Leave One Out Cross Validation The Accuarcy has come 70.25%